mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/processing/building/impl/PojoIndexingProcessorTypeNodeBuilder.java

```diff
@@ -135,13 +140,16 @@ public void closeOnFailure() {
 					.map( Optional::get )
 					.forEach( immutablePropertyNodes::add );
 
-			if ( parentIndexObjectReferences.isEmpty() && immutableBridgeHolders.isEmpty() && immutablePropertyNodes
-					.isEmpty() ) {
-				/*
-				 * If this node doesn't create any object in the document, and it doesn't have any bridge,
-				 * nor any property node, then it is useless and we don't need to build it.
-				 */
-				return Optional.empty();
+			if ( immutableBridgeHolders.isEmpty() && immutablePropertyNodes.isEmpty() ) {
+				if ( parentIndexObjectReferences.isEmpty() ) {
+					/*
+					 * If this node doesn't create any object in the document, and it doesn't have any bridge,
+					 * nor any property node, then it is useless and we don't need to build it.
+					 */
+					return Optional.empty();
+				}
+
+				throw log.invalidIndexedEmbedded( modelPath.getParent().getParent().getPropertyModel().getName(), modelPath.getTypeModel().getName() );
```

Comment:
This is not the right place for this error. I explicitly discussed the right way to implement this fix with you, and I mentioned why we needed to do it that way. I guess I wasn't clear enough.

Fields can be filtered out. Consider this:

```
@Entity 
@Indexed
public class Level1 {
    @IndexedEmbedded(includePaths = "level3b.text")
    private Level2 level2;
}

@Entity 
@Indexed
public class Level2 {
    @IndexedEmbedded
    private Level3A level3a;

    @IndexedEmbedded
    private Level3B level3b;
}

@Entity 
public class Level3A {
    @GenericField
    private String text;
}

@Entity 
public class Level3B {
    @GenericField
    private String text;
}
```

When building the indexing processor for `Level1`, it's entirely possible that a type node is created for `level3a`, but it won't define any field, because the `includePaths` clause filters it out.
Now this specific setup might not fail because there are optimizations in place which may trigger an early stop and prevent the type node from being created, but you get the idea: I'm sure there are other cases where an empty type node can legitimately be created.

The right way to address this problem, as we discussed previously, is to:

1. Replace `org.hibernate.search.engine.mapper.mapping.building.spi.TypeMetadataContributorProvider#forEach` with a method that takes a type model as a paremeter, and returns a set of `C`. Essentially `Set<C> get(MappableTypeModel typeModel)`.
2. In `org.hibernate.search.mapper.pojo.processing.building.impl.PojoIndexingProcessorValueNodeBuilderDelegate#indexedEmbedded`, test the returned set before doing anything: if it's empty, it means there isn't any mapping defined for the embedded type, and we should throw the exception.

Reply:
Sorry if I did remember that. Actually I don't have a good memory.
Not a problem to move the place where to address it.

Just for my very curiosity. How does it possible that:
> it won't define any field, because the includePaths clause filters it out.

if `Level3B` defines the generic field `text`?
Thanks ;)

Reply:
@yrodiere sorry. I'm trying to apply the strategy I think you pointed out here.

Probably I didn't grasp something. But the list is not empty, even if we have a String as a nested model type.

My attempt resides in the commit named:
`HSEARCH-899 Move check to PojoIndexingProcessorValueNodeBuilderDelegate`.

Thanks

Reply:
Looks like we solved this.

Regarding your question:

> Just for my very curiosity. How does it possible that:
>
> >    it won't define any field, because the includePaths clause filters it out.
>
> if Level3B defines the generic field text?

It's `level3a` which may end up empty, because it is not mentioned in `includePaths`. `level3b` and `level3b.text` will definitely be included, since they are explicitly mentioned in `includePaths`.